From 2d8a07e976859c7e8582ed8f4d7849f7e8b8a68c Mon Sep 17 00:00:00 2001
From: Jean-Christophe Fillion-Robin <jchris.fillionr@kitware.com>
Date: Mon, 6 Nov 2017 09:56:43 -0500
Subject: [PATCH 1/3] Prevent incorrect include of "io.h" and "memory.h" found
 in libmpdec directory

This commit improves support for building python with built-in extensions
on windows where the header provided by libmpdec would be included instead
of the system one.
---
 Modules/_decimal/libmpdec/basearith.c                | 2 +-
 Modules/_decimal/libmpdec/io.c                       | 4 ++--
 Modules/_decimal/libmpdec/memory.c                   | 2 +-
 Modules/_decimal/libmpdec/{io.h => mpd_io.h}         | 0
 Modules/_decimal/libmpdec/{memory.h => mpd_memory.h} | 0
 Modules/_decimal/libmpdec/mpdecimal.c                | 2 +-
 6 files changed, 5 insertions(+), 5 deletions(-)
 rename Modules/_decimal/libmpdec/{io.h => mpd_io.h} (100%)
 rename Modules/_decimal/libmpdec/{memory.h => mpd_memory.h} (100%)

diff --git a/Modules/_decimal/libmpdec/basearith.c b/Modules/_decimal/libmpdec/basearith.c
index 35de6b8..c0953a3 100644
--- a/Modules/_decimal/libmpdec/basearith.c
+++ b/Modules/_decimal/libmpdec/basearith.c
@@ -32,7 +32,7 @@
 #include <string.h>
 #include <assert.h>
 #include "constants.h"
-#include "memory.h"
+#include "mpd_memory.h"
 #include "typearith.h"
 #include "basearith.h"
 
diff --git a/Modules/_decimal/libmpdec/io.c b/Modules/_decimal/libmpdec/io.c
index a45a429..f517264 100644
--- a/Modules/_decimal/libmpdec/io.c
+++ b/Modules/_decimal/libmpdec/io.c
@@ -37,9 +37,9 @@
 #include <locale.h>
 #include "bits.h"
 #include "constants.h"
-#include "memory.h"
+#include "mpd_memory.h"
 #include "typearith.h"
-#include "io.h"
+#include "mpd_io.h"
 
 
 /* This file contains functions for decimal <-> string conversions, including
diff --git a/Modules/_decimal/libmpdec/memory.c b/Modules/_decimal/libmpdec/memory.c
index 0f41fe5..f4140b2 100644
--- a/Modules/_decimal/libmpdec/memory.c
+++ b/Modules/_decimal/libmpdec/memory.c
@@ -30,7 +30,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "typearith.h"
-#include "memory.h"
+#include "mpd_memory.h"
 
 
 /* Guaranteed minimum allocation for a coefficient. May be changed once
diff --git a/Modules/_decimal/libmpdec/io.h b/Modules/_decimal/libmpdec/mpd_io.h
similarity index 100%
rename from Modules/_decimal/libmpdec/io.h
rename to Modules/_decimal/libmpdec/mpd_io.h
diff --git a/Modules/_decimal/libmpdec/memory.h b/Modules/_decimal/libmpdec/mpd_memory.h
similarity index 100%
rename from Modules/_decimal/libmpdec/memory.h
rename to Modules/_decimal/libmpdec/mpd_memory.h
diff --git a/Modules/_decimal/libmpdec/mpdecimal.c b/Modules/_decimal/libmpdec/mpdecimal.c
index 593f9f5..2fd8b33 100644
--- a/Modules/_decimal/libmpdec/mpdecimal.c
+++ b/Modules/_decimal/libmpdec/mpdecimal.c
@@ -36,7 +36,7 @@
 #include "bits.h"
 #include "convolute.h"
 #include "crt.h"
-#include "memory.h"
+#include "mpd_memory.h"
 #include "typearith.h"
 #include "umodarith.h"
 
-- 
2.5.0

